### PR TITLE
chore(template): get rid of extra <div /> wrapping the Template if possible

### LINKED ISSUE #422: Too many wrapping <div>
While building the documentation I discovered a few extra `div`s (mostly wrapping the `<Template />` elements) that get can get rid of. It simplifies a lot the understanding of the generated HTML.

### PR DESCRIPTION
**Pull request by redox**
Fix #422

BREAKING CHANGE: some extra divs have been removed and my break your CSS rules.

### CODE REVIEW

File: components/Slider/__tests__/Slider-test.js [hunk 1]
```diff
@@ -28,17 +28,15 @@ describe('Slider', () => {
...
+      <Nouislider
+        animate={false}
+        behaviour="snap"
+        connect
+        cssClasses={{}}
+        cssPrefix="ais-range-slider--"
+        onChange={() => {}}
+        templateProps={{}}
+      />
     );
```
Comment (redox) [inline components/Slider/__tests__/Slider-test.js]: @vvo I've also removed that one; any idea why you put it there?

Comment (vvo) [inline components/Slider/__tests__/Slider-test.js]: No idea damn brain

**Comment by vvo** [comment]
I guess you tried the UI in various ways (results, no results) etc.. and it worked?

**Comment by vvo** [comment]
LGTM

File: components/__tests__/Template-test.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ describe('Template', () => {
     renderer.render(<Template {...props} />);
     let out = renderer.getRenderOutput();
 
-    expect(out).toEqualJSX(<div dangerouslySetInnerHTML={{__html: 'it works with strings'}}></div>);
+    expect(out).toEqualJSX(<div className={undefined} dangerouslySetInnerHTML={{__html: 'it works with strings'}}></div>);
```
Comment (pixelastic) [inline components/__tests__/Template-test.js]: It's weird we have to test that. Shouldn't the rclassName` just not be here?

cc. @vvo

File: components/Hits.js [hunk 3]
```diff
@@ -7,13 +7,13 @@ class Hits extends React.Component {
   renderWithResults() {
     let renderedHits = map(this.props.results.hits, hit => {
       return (
-        <div className={this.props.cssClasses.item} key={hit.objectID}>
```
Comment (vvo) [inline components/Hits.js]: Ah now I think I remember, the key should not be put on the Template directly but on a wrapping div, I think that's a react requirement.

Comment (vvo) [inline components/Hits.js]: No my bad, should be ok: http://facebook.github.io/react/docs/multiple-components.html#dynamic-children the example is using the same notation than us

File: components/__tests__/Template-test.js (hunk 2 above)
Comment (vvo) [inline components/__tests__/Template-test.js]: Yes I was too fast on this, @redox there's no "cssClass" prop on template (also present in documentation). SO I wonder what are we doing here?

Comment (vvo) [inline components/__tests__/Template-test.js]: Sorry there's one now.

Comment (vvo) [inline components/__tests__/Template-test.js]: But the className should be "custom-root custom-empty"?

Comment (redox) [inline components/__tests__/Template-test.js]: Hmmmm, no; why should it be `"custom-root custom-empty"` ? We're in the template-tests here.

Comment (vvo) [inline components/__tests__/Template-test.js]: right :door:

Comment (pixelastic) [inline components/__tests__/Template-test.js]: But still, what is the difference between testing `<div className={undefined}>` and testing `<div>`? I would expect a props with a value of `undefined` to just not be in the resulting JSX

Comment (vvo) [inline components/__tests__/Template-test.js]: That's just how we implemented it but `className={undefined}` is not the same as no className. Like `o = {a: undefined}` is not the same as `o = {}`

Comment (vvo) [inline components/__tests__/Template-test.js]: Still it looks weird yes in the tests agree

**Pull request merged**